packages/flutter/test/painting/image_cache_test.dart

```diff
@@ -121,14 +121,14 @@ void main() {
       expect(imageCache.currentSizeBytes, 256);
     });
 
-    test('Increases cache size if an image is loaded that is larger then the maximum size', () async {
+    test('Image larger than maximumSizeBytes', () async {
       const TestImage testImage = TestImage(width: 8, height: 8);
 
       imageCache.maximumSizeBytes = 1;
       await extractOneFrame(const TestImageProvider(1, 1, image: testImage).resolve(ImageConfiguration.empty));
-      expect(imageCache.currentSize, 1);
-      expect(imageCache.currentSizeBytes, 256);
-      expect(imageCache.maximumSizeBytes, 256 + 1000);
+      expect(imageCache.currentSize, 0);
+      expect(imageCache.currentSizeBytes, 0);
+      expect(imageCache.maximumSizeBytes, 1);
```

Comment:
We should have a test that checks the exception we throw into `FlutterError.onError` for this.

Reply:
As written, this test currently emits:

```
00:08 +3: ImageCache Increases cache size if an image is loaded that is larger then the maximum size                                                                                                                                                                                                                    
══╡ EXCEPTION CAUGHT BY IMAGE RESOURCE SERVICE ╞════════════════════════════════════════════════════
The following message was thrown by a synchronously-called image listener:
Attempting to cache an image whose size exceeds ImageCache.maximumSizeBytes: 1

When the exception was thrown, this was the stack:
#0      ImageCache.putIfAbsent.listener (package:flutter/src/painting/image_cache.dart:174:9)
#1      ImageStreamCompleter.addListener (package:flutter/src/painting/image_stream.dart:358:18)
#2      ImageCache.putIfAbsent (package:flutter/src/painting/image_cache.dart:189:14)
#3      ImageProvider.resolve.<anonymous closure>.<anonymous closure> (package:flutter/src/painting/image_provider.dart:325:84)
(elided 18 frames from package dart:async and package stack_trace)
════════════════════════════════════════════════════════════════════════════════════════════════════
00:08 +9: All tests passed!          
```

I'm a bit surprised we don't consider that a failure in the test, but we probably should, and we should at elast be asserting it happens if it's what we want.